Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -3443,10 +3443,15 @@ RefPtr<CSSValue> ComputedStyleExtractor::valueForPropertyInStyle(const RenderSty
         return createConvertingToCSSValueID(style.inputSecurity());
     case CSSPropertyLeft:
         return positionOffsetValue(style, CSSPropertyLeft, renderer);
-    case CSSPropertyLetterSpacing:
-        if (!style.letterSpacing())
-            return CSSPrimitiveValue::create(CSSValueNormal);
-        return zoomAdjustedPixelValue(style.letterSpacing(), style);
+    case CSSPropertyLetterSpacing: {
+        const Length& spacing = style.computedLetterSpacing();
+        if (spacing.isFixed()) {
+            if (spacing.isZero())
+                return CSSPrimitiveValue::create(CSSValueNormal);
+            return zoomAdjustedPixelValue(spacing.value(), style);
+        }
+        return CSSPrimitiveValue::create(spacing, style);
```

Comment:
The spec says it should always compute to an absolute length. Is it what we're doing? Do we have tests that cover this? 

It sounds to me that what we want is to return the a pixel length containing the raw float value

Reply:
No, the percentages need to be preserved in the computed value, otherwise they don't inherit correctly.

> Computed value: absolute length and/or percentage

And since there's no legacy compat constraints on percentages here, `getComputedStyle()` doesn't have a special exception (like `width` does) to return the used value instead of the computed value.

Reply:
Ah I was reading CSS text 3 instead of 4, sorry.